### ISSUE TITLE
tracking?

### ISSUE THREAD

**Issue by robgpeak** [robgpeak/CL-WebExtension#28]
Hi @AddisonFreeman and @cgladue do we currently have a way to track how many times the casino's logo shows up in a google search when they have the plug in installed?

**Comment by AddisonFreeman** [comment]
Not currently, but that wouldn't be too much effort. Just send a count to
the backend on every search result page for that user. The backend would
keep a running total.

**Comment by robgpeak** [comment]
@cgladue ???

**Comment by robgpeak** [comment]
@cgladue is this something you would need to do on your end?

**Comment by cgladue** [comment]
umm, yea i can do this, ill send you and addison something with the specs tonight.

**Comment by robgpeak** [comment]
Thanks Chad.  and would it be a separate issue (that I should set up here in Github) if we wanted to see who has actually installed the extension at any given time?  That would be an attribute I would like to see in Intercom so we could do email messaging based on whether they do or do not have the plug in installed...

**Comment by cgladue** [comment]
ummm, i think i might be able to record that data with the way things are now without any work on addisons side ...  so yea create a new issue for that too

**Comment by cgladue** [comment]
@robgpeak can this just be a single field in the database that increments each time the partner shows up in search results ?

**Comment by robgpeak** [comment]
So is what you are asking that the image below would show in our back end that "Xclub" had 4 impressions...?  Can we also count clicks?
![screen shot 2018-09-18 at 3 42 56 pm](https://user-images.githubusercontent.com/28393235/45712303-f8799080-bb59-11e8-98e0-ee574ce4acda.png)

**Comment by cgladue** [comment]
@AddisonFreeman @robgpeak 

i was actually able to do this on my end automatically without anything needed from addison, since he is querying me with the checkDomain call for each domain on the search results page, i am just tracking the results that the partner is showing up for automatically if "isAdvertiser = true"

**Comment by cgladue** [comment]
@robgpeak right now its not visible anywhere, but the data is being collected.  would like to hear your recommendation on where in the backend you want to display it.

**Comment by robgpeak** [comment]
On the https://shop.complinks.co/dashboard/partner/index/12 page (for each individual partner of course, so they can see their own data) underneath the Advertiser View Totals, we would have another section called, "Brand Impressions" and it would show Date and the Total Brand Impressions (the number of times their logo was seen for that day as a result of a search through the extension.)

It would also be it's own separate section "Brand Impressions" in the left nav under the partner...  The icon is https://material.io/tools/icons/?icon=find_in_page&style=baseline

The ultimate objective is to use this as an additional value proposition to casinos to show them that they will be getting free advertising/branding when people install the extension and see their logo along side the listings...

<img width="1123" alt="screen shot 2018-09-18 at 5 38 40 pm" src="https://user-images.githubusercontent.com/28393235/45718108-f4556f00-bb69-11e8-8293-d543460c6770.png">

**Comment by cgladue** [comment]
@robgpeak i actually track which advertisers the partner impression shows up for in search results ... so i could show the advertiser name as well ...  do you think they would want to break it down by the most popular advertiser in search results ?

**Comment by robgpeak** [comment]
Yeah, that would be great, but I would also like to be able to see totals for the days.  So how about showing the date and then the total brand impressions as a hyperlink and when someone click on it, it would show the breakdown by advertisers.

**Comment by cgladue** [comment]
@robgpeak @AddisonFreeman actually i am not able to do it without you making some changes on the extension side...

here is the JSON for the request, send in an array of advertiserIds that were displayed in the search result, preferable 1 request per page load with all ids in the 1 request ...

POST /api/v1/recordSearchImpression
```
{
	"advertiserIds" : [
		5464, 1234, 5424, 2555	
	]
}
```

**Comment by cgladue** [comment]
@robgpeak How about this:

![image](https://user-images.githubusercontent.com/7914133/45724250-38a13900-bb83-11e8-80dd-e335b05f19d2.png)


![image](https://user-images.githubusercontent.com/7914133/45724263-43f46480-bb83-11e8-9949-7320f4c307f7.png)


![image](https://user-images.githubusercontent.com/7914133/45724301-79994d80-bb83-11e8-808e-00d83339a10c.png)

**Comment by robgpeak** [comment]
Looks good

**Comment by robgpeak** [comment]
@cgladue Does this show up in real time?  I just went to a google search and saw the Xclub logo above four different advertisers, but when I look in https://xclub.complinks.co/dashboard/partner/brandimpressions I don't see anything except the first four that were in there from the 18th.

**Comment by robgpeak** [comment]
@cgladue oops, I just re-read your message that you need Addison to do something with the JSON.  @AddisonFreeman please hit me up on skype to chat about this.

**Comment by cgladue** [comment]
@robgpeak yea, the problem was addison is calling the "CheckDomain" call on every page load even if its not on a search result, so if i do it without him makeing somechanges it would increment even if they went directly to the advertisers page and not a search, which i dont think you want it to do right ?

**Comment by robgpeak** [comment]
@cgladue but if they went directly to an advertiser's page (not through a search), and _**did have**_ the extension installed, the user _**would**_ see the casino partner's logo in the pop up (below), yes?  If that is the case, I think that too should count as a brand impression...
<img width="690" alt="screen shot 2018-09-20 at 11 13 25 am" src="https://user-images.githubusercontent.com/28393235/45828343-4f9c7400-bcc6-11e8-8d5a-35fb587ed49d.png">

**Comment by cgladue** [comment]
ok, if thats the case i can implement it without anything additional on addisons side...

**Comment by cgladue** [comment]
you just wont know the # of search impressions vs the number of direct impressions

**Comment by robgpeak** [comment]
I think that will suffice for now

**Comment by robgpeak** [comment]
I wanted to find out if we can break down the number of impressions separately from the number of searches?  Meaning, if someone does a search for shoes, and three advertisers show up with the casino partner's logo over it, that would be three impressions, but only one search...

**Comment by robgpeak** [comment]
Also for the brand impressions, it would be great to see in charts like to have on the partner dashboard.  It will look sexier when we show it to casinos.

**Comment by cgladue** [comment]
currently no for breaking them down, i could add another field for count, but without addison that count would also increment when they just went directly to the advertisers page, wouldnt be just for searches

**Comment by cgladue** [comment]
charts are doable....

**Comment by robgpeak** [comment]
Ok just do the charts for now and the other stuff we previously discussed.

**Comment by cgladue** [comment]
where would you like the chart, under the dashboard, or under the Brand Impressions option ?

**Comment by robgpeak** [comment]
Can you put it in both?

**Comment by cgladue** [comment]
@robgpeak how about this:

https://xclub.complinks.co/dashboard/partner

https://xclub.complinks.co/dashboard/partner/brandimpressions

**Comment by robgpeak** [comment]
Yeah that's fine.  When will it be populated with data?

**Comment by cgladue** [comment]
@robgpeak should do it starting now.

**Comment by robgpeak** [comment]
If it's not too much trouble, please have the links from the impressions column show the details of just the date from the link that was clicked on.  Meaning, in xclub now, if you click on either of those links, it brings you to the details=true page, but it would be a better UX if it just showed the details for the day that corresponds to the date someone clicked on.

**Comment by cgladue** [comment]
@robgpeak how about now

**Comment by robgpeak** [comment]
cool.  Thanks!

**Comment by robgpeak** [comment]
@cgladue and @AddisonFreeman Just looking at todays numbers 9/24/18 and I see 14 branded impressions, but when I drill in, I see they are all Verizon Wireless which seems very strange.  I wanted to evaluate what it would take to break it down even further, meaning down to the individual user just to get a little more granular.

**Comment by cgladue** [comment]
@robgpeak i can add user without needing @AddisonFreeman

**Comment by robgpeak** [comment]
OK.  Let's see what it looks like.  I think we should have the number of impressions in the last column on the page https://xclub.complinks.co/dashboard/partner/brandimpressions?detail=true&startDate=09/24/2018&endDate=09/24/2018 be a link to 'brandimpressions/users' page or something like it which would show the Account ID, first name, last name, store brand impression seen.
